mldsa/polyvec.c

```diff
@@ -39,13 +39,33 @@ void polyvec_matrix_pointwise_montgomery(polyveck *t,
 /**************************************************************/
 
 void polyvecl_uniform_eta(polyvecl *v, const uint8_t seed[MLDSA_CRHBYTES],
-                          uint16_t nonce)
+                          const uint16_t nonce)
 {
   unsigned int i;
-
+  uint16_t n = nonce;
   for (i = 0; i < MLDSA_L; ++i)
+  __loop__(
+    assigns(i, n, object_whole(v))
+    invariant(i <= MLDSA_L)
+    invariant(forall(k1, 0, i,
+      array_abs_bound(v->vec[k1].coeffs, 0, MLDSA_N, MLDSA_ETA + 1))))
   {
-    poly_uniform_eta(&v->vec[i], seed, nonce++);
+    poly t;
+    poly_uniform_eta(&t, seed, n);
+
+#ifdef CBMC
+#pragma CPROVER check push
+#pragma CPROVER check disable "unsigned-overflow"
+#endif
+    n++;
+#ifdef CBMC
+#pragma CPROVER check pop
+#endif
```

Comment:
As in #229, this should be replaced by a precondition bound on `nonce`.

Reply:
Ignore this, this PR is just staged ontop of the old one that has now been updated. This will be removed when I rebase

Reply:
Ah right, this was from the old PR -- I misread!